src/web/client/test/integration/errorHandler.test.ts

```diff
@@ -0,0 +1,634 @@
+        //Action
+        const result = checkMandatoryPathParameters(appName, entity, entityId);
+        //Assert
+        expect(result).true;
+    });
+
+    it("checkMandatoryPathParameters_shouldReturnFalseWhenEntityIsBlank", () => {
+        //Act
+        const _mockShowErrorDialog = spy(vscode.window, "showErrorMessage");
+        const _mockSendErrorTelemetry = spy(
+            powerPlatformExtensionContext.telemetry,
+            "sendErrorTelemetry"
+        );
+        const appName = "portal";
+        const entity = "";
+        const entityId = "3";
+        //Action
+        const result = checkMandatoryPathParameters(appName, entity, entityId);
+        //Assert
+        const detailMessage = localize(
+            "microsoft-powerapps-portals.webExtension.init.workspace.error.desc",
+            "Check the URL and verify the parameters are correct"
+        );
+        const options = { detail: detailMessage, modal: true };
+        expect(result).false;
+        assert.calledOnceWithExactly(
+            _mockShowErrorDialog,
+            localize(
+                "microsoft-powerapps-portals.webExtension.init.workspace.error",
+                "There was a problem opening the workspace"
+            ),
+            options
+        );
+        assert.calledOnceWithExactly(
+            _mockSendErrorTelemetry,
+            telemetryEventNames.WEB_EXTENSION_MANDATORY_PATH_PARAMETERS_MISSING
+        );
+    });
+
+    it("checkMandatoryPathParameters_shouldReturnFalseWhenEntityIdIsBlank", () => {
+        //Act
+        const _mockShowErrorDialog = spy(vscode.window, "showErrorMessage");
+        const _mockSendErrorTelemetry = spy(
+            powerPlatformExtensionContext.telemetry,
+            "sendErrorTelemetry"
+        );
+        const appName = "portal";
+        const entity = "entity";
+        const entityId = "";
+        //Action
+        const result = checkMandatoryPathParameters(appName, entity, entityId);
+        //Assert
+        const detailMessage = localize(
+            "microsoft-powerapps-portals.webExtension.init.workspace.error.desc",
+            "Check the URL and verify the parameters are correct"
+        );
+        const options = { detail: detailMessage, modal: true };
+        expect(result).false;
+        assert.calledOnceWithExactly(
+            _mockShowErrorDialog,
+            localize(
+                "microsoft-powerapps-portals.webExtension.init.workspace.error",
+                "There was a problem opening the workspace"
+            ),
+            options
+        );
+        assert.calledOnceWithExactly(
+            _mockSendErrorTelemetry,
+            telemetryEventNames.WEB_EXTENSION_MANDATORY_PATH_PARAMETERS_MISSING
+        );
+    });
+
+    it("checkMandatoryPathParameters_shouldReturnFalseWhenPppNameIsDifferentFromPortal", () => {
+        //Act
+        const _mockShowErrorMessage = stub(vscode.window, "showErrorMessage");
+        const appName = "por";
```

Comment:
What happens when appname is undefined. I think we should have test case around that also

Reply:
Cannot convert string to undefined